src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -107,6 +107,68 @@ public static SymbolVisibility GetResultantVisibility(this ISymbol symbol)
 
             return visibility;
         }
+
+        public static bool MatchMemberDerivedByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType.DerivesFrom(type) && member.MetadataName == name;
+        }
+
+        public static bool MatchMethodDerivedByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMemberDerivedByName(type, name);
+        }
+
+        public static bool MatchMethodByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Method && member.MatchMemberByName(type, name);
+        }
+
+        public static bool MatchPropertyDerivedByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMemberDerivedByName(type, name);
+        }
+
+        public static bool MatchFieldDerivedByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Field && member.MatchMemberDerivedByName(type, name);
+        }
+
+        public static bool MatchMemberByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.ContainingType == type && member.MetadataName == name;
+        }
+
+        public static bool MatchPropertyByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Property && member.MatchMemberByName(type, name);
+        }
+
+        public static bool MatchFieldByName(this ISymbol member, INamedTypeSymbol type, string name)
+        {
+            return member != null && member.Kind == SymbolKind.Field && member.MatchMemberByName(type, name);
+        }
+
+        public static ITypeSymbol GetVariableSymbolType(this ISymbol symbol)
```

Comment:
A lot of code in this PR is just duplicate from #705. Since #705 is already merged, could you please sync and resolve the conflict so duplicate code wouldn't show up in this PR?